### PR TITLE
Add attempt to connect to socket before selecting it as runtime

### PR DESCRIPTION
**Pull request by eharris128**
What
===============
- Ping sockets before designating it as runtime

Why
===============
- crt.AutoRuntime would tell the `tui -> i` images `OnCommand` invocation to use `containerd` as runtime. However, the application could not talk to this containerd socket due to permission issues. The socket was there from the perspective of `Stat` - but this did not guarantee sufficient access to communicate with the socket.

How Tested
===============
mint images
mint tui -> i

## New Experience
- When a user's environment has podman & containerd, but the application does not have sufficient permissions to access these sockets, then fallback to docker:
image

### CODE REVIEW

File: pkg/crt/runtimes.go [hunk 1]
```diff
@@ -106,11 +113,12 @@ func AvailableRuntimes() []string {
 
 func AutoSelectRuntime() string {
 	available := AvailableRuntimes()
+	log.Debugf("Available runtimes: %v", available)
 	if len(available) > 0 {
 		return available[0]
 	}
 
-	return DockerRuntime
+	return DockerRuntime // Question -> This runtime may not necessarily be available?
```
Comment (kcq) [inline pkg/crt/runtimes.go:121]: Yes, it is a good idea to check if the default/assumed runtime (when no explicit runtime param is specified) is available. A wishlist enhancement :-)

**Comment by kcq** [comment]
@eharris128 there might be another bug to investigate there... The "auto crt" logic shouldn't have picked containerd unless containerd is the only thing is installed. What's the docker socket you have in your environment? What's the docker context info?

File: pkg/crt/runtimes.go (hunk 1 above)
Comment (eharris128) [reply pkg/crt/runtimes.go:121]: A future TODO

**Comment by eharris128** [comment]
` What's the docker socket you have in your environment? What's the docker context info?` 

Will share these details next time I am on my dev machine.

**Comment by eharris128** [comment]
@kcq 

image



image

**Comment by kcq** [comment]
@eharris128 can you also run `docker context ls` and `ls -lh /var/run/docker.sock`

**Comment by kcq** [comment]
Either way, the `/var/run/docker.sock` socket should have been picked by the auto container runtime selection logic. With the Docker runtime detection it's supposed to go through these three stages:

1. Try using the unix socket from docker context
2. If there's no docker context configured try using /var/run/docker.sock
3. Try using the user directory located docker socket (usually from Docker Desktop installs)

**Comment by eharris128** [comment]
@kcq very delayed here, but I ruined the reproducibility as I upgraded to Ubuntu 24 LTS last week (in order to work with Podman - as the Ubuntu 22 LTS's installation of Podman does not work for mint).

With the Ubuntu 24 LTS default docker configuration, we no longer get the issue of defaulting to containerd (when docker is installed).

**Comment by kcq** [comment]
@eharris128 no worries :) can you create a GitHub issue with the notes from our discussion about it

**Pull request merged**